stdlib/universe/filter.go

```diff
@@ -359,3 +361,42 @@ func (RemoveTrivialFilterRule) Rewrite(ctx context.Context, filterNode plan.Node
 	anyNode := filterNode.Predecessors()[0]
 	return anyNode, true, nil
 }
+
+// MergeFiltersRule merges Filter nodes whose body is a single return to create one Filter node.
+type MergeFiltersRule struct{}
+
+func (MergeFiltersRule) Name() string {
+	return "MergeFiltersRule"
+}
+
+func (MergeFiltersRule) Pattern() plan.Pattern {
+	return plan.Pat(FilterKind, plan.Pat(FilterKind, plan.Any()))
+}
+
+func (MergeFiltersRule) Rewrite(ctx context.Context, filterNode plan.Node) (plan.Node, bool, error) {
+	// conditions
+	filterSpec1 := filterNode.ProcedureSpec().(*FilterProcedureSpec)
+	bodyExpr1, ok := filterSpec1.Fn.Fn.GetFunctionBodyExpression()
+	if !ok {
+		// Not an expression.
+		return filterNode, false, nil
+	}
+	filterSpec2 := filterNode.Predecessors()[0].ProcedureSpec().(*FilterProcedureSpec)
+	bodyExpr2, ok := filterSpec2.Fn.Fn.GetFunctionBodyExpression()
+	if !ok {
+		// Not an expression.
+		return filterNode, false, nil
+	}
+	if filterSpec1.KeepEmptyTables != filterSpec2.KeepEmptyTables && !filterSpec2.KeepEmptyTables {
```

Comment:
Can you add a comment the describes the reasoning of the logic here? Just so it's clear if we ever need to revisit this.